src/Resources/Grapher/FactorsResource.php

```diff
@@ -6,58 +6,35 @@
 
 namespace SphereMall\MS\Resources\Grapher;
 
-
-use SphereMall\MS\Entities\Entity;
-use SphereMall\MS\Exceptions\MethodNotFoundException;
 use SphereMall\MS\Resources\Resource;
+use SphereMall\MS\Resources\Traits\DetailResource;
 
 /**
  * Class FactorsResource
+ *
  * @package SphereMall\MS\Resources\Grapher
  */
 class FactorsResource extends Resource
 {
+    use DetailResource;
+
     #region [Override methods]
     public function getURI()
     {
         return "factors";
     }
-    #endregion
 
     /**
-     * Get list of entities
+     * @param $factorCode
      *
-     * @param null|int|string $param
-     *
-     * @return Entity|Entity[]
+     * @return array|int|\SphereMall\MS\Entities\Entity|\SphereMall\MS\Lib\Collection
      * @throws \GuzzleHttp\Exception\GuzzleException
-     * @deprecated
      */
-    public function full($param = null)
-    {
-        $uriAppend = 'full';
-        $params    = $this->getQueryParams();
-
-        if (!is_null($param)) {
-            $uriAppend = $uriAppend . "/$param";
-        }
-
-        $response = $this->handler->handle('GET', false, $uriAppend, $params);
-
-        return $this->make($response);
-    }
-
-    public function detail($factorId)
-    {
-        $response = $this->handler->handle('GET', false, "detail/code/{$factorId}");
```

Comment:
detail/code/$factorId or detail/url/$factorId is correct one?
As far i know, DetailResource trait has second variant.

Reply:
As far you can see that this part of code was deleted ;)